### PR TITLE
add checks for valid property values for ClutchedPathSpring and PathSpring classes

### PR DESCRIPTION
**Pull request by carmichaelong**
Fixes issue #1607

### Brief summary of changes
Ensures that property values are valid through `extendFinalizeFromProperties()`. Will throw an exception if any are not valid. Note that the issue mentioned also noted `CoordinateLimitForce` but it appeared to me that that class initializes all its properties to a valid number, and that the coordinate name was checked to exist as well.

### Testing I've completed
All tests pass locally on a Windows x32 build.

### Looking for feedback on...
- The checks are sufficient and cover all properties needed.
- Doxygen comment for the default constructors.

### CHANGELOG.md (choose one)
no need to update: just a bug fix

### CODE REVIEW

File: OpenSim/Simulation/Model/PathSpring.h [hunk 1]
```diff
@@ -82,7 +82,9 @@ OpenSim_DECLARE_CONCRETE_OBJECT(PathSpring, Force);
 // PUBLIC METHODS
 //==============================================================================
     
-    /** Default constructor */
+    /** Construct a ClutchedPathSpring with default parameters. Users should
```
Comment (tkuchida) [inline OpenSim/Simulation/Model/PathSpring.h]: ClutchedPathSpring -> PathSpring

File: OpenSim/Actuators/ClutchedPathSpring.cpp [hunk 2]
```diff
@@ -122,6 +122,19 @@ void ClutchedPathSpring::setInitialStretch(double stretch0)
      set_initial_stretch(getStretch(state));
  }
  
+ void ClutchedPathSpring::extendFinalizeFromProperties()
+ {
+     Super::extendFinalizeFromProperties();
+
+     OPENSIM_THROW_IF_FRMOBJ(get_stiffness() < 0,
```
Comment (tkuchida) [inline OpenSim/Actuators/ClutchedPathSpring.cpp]: One more concern: will these exceptions really be thrown if the property values are `SimTK::NaN`? IIRC, `SimTK::NaN < 0` evaluates to `false`. Might need to do something like
```cpp
OPENSIM_THROW_IF_FRMOBJ(SimTK::isNaN(get_stiffness()) || get_stiffness() < 0,
                        ...);
```

**Comment by tkuchida** [comment]
:+1: Great! Please merge when tests :white_check_mark: (or :hourglass:).

**Comment by carmichaelong** [comment]
The one failure was due to timeout. Merging. Thanks @tkuchida and @chrisdembia!

**Pull request merged**